### ISSUE TITLE
Google-fonts should not allow duplicates to be enqueued

### ISSUE THREAD

**Issue by gerardbm** [valendesigns/option-tree#470]
The new `google-fonts` option inserts a meta link into the `head` even if this option is not used. The ID of the meta link is `ot-google-fonts-css`, and it has de 'Open Sans' font as default value.

```html
<link rel='stylesheet' id='ot-google-fonts-css'  href='//fonts.googleapis.com/css?family=Open+Sans:300,300italic,regular,italic,600,600italic' type='text/css' media='all' />
```

Is there a way to remove this extra link?

---------------------
Actually, I have an another little question about this subject. I've been using the Google Fonts since some months ago thanks to a filter that overwrites the default OT fonts:

```php
add_filter('ot_recognized_font_families', 'filter_ot_recognized_font_families', 10, 2);
```
This is interesting because it mixes the `typography` option with Google Fonts, allowing the user to edit every CSS parameter of a font from the same place. The inconvenient of Google Fonts is the loading time. When OT is loading the Theme Options page, it has to load a huge array of fonts, so it's slower than before (a lot). I tested it with my own array and using the new `google-fonts` option type, by separate, and this happens in both cases. At the moment I limited the use of this array only in 3 options of my theme (logo text, body font, headings), but it'd be great if OT loaded it faster in the future, such as other frameworks do (I don't know how).

Thanks for your time and good job.

Regards,

**Comment by gerardbm** [comment]
The meta-link is removed after a load of the Theme Options page. I didn't see it this morning. There was no issues here.

But I keep the second part of my comment.

Also, I have a little question about the meta-link. When you use the `google-fonts` option some times (three, for example), if a Google Font is repeated, the font-family will be added ```n``` times in the meta-link. Anyway, the content of this link is the same, so I don't know if I should worry about it. The meta-link would be:

```html
<link rel='stylesheet' id='ot-google-fonts-css'  href='//fonts.googleapis.com/css?family=Open+Sans:300,300italic,regular,italic,600,600italic|Open+Sans:300,300italic,regular,italic,600,600italic|Open+Sans:300,300italic,regular,italic,600,600italic' type='text/css' media='all' />
```
The following link has the same content:

```html
<link rel='stylesheet' id='ot-google-fonts-css'  href='//fonts.googleapis.com/css?family=Open+Sans:300,300italic,regular,italic,600,600italic' type='text/css' media='all' />
```

It simply repeats the part «Open+Sans:300,300italic,regular,italic,600,600italic» three times.

I developed a solution to this, but, as I said, I don't know if it's something to worry or not.

```php
function google_fonts_load() {

	$typography_1 = ot_get_option('typography_1', array("font-family" => "Open Sans"));
	$typography_2 = ot_get_option('typography_2', array("font-family" => "Open Sans"));
	$typography_3 = ot_get_option('typography_3', array("font-family" => "Open Sans"));

	$input = array(
	urlencode($typography_1['font-family']),
	urlencode($typography_2['font-family']),
	urlencode($typography_3['font-family'])
	);

	$result = array_unique($input);

	$fonts_list = array();
	foreach ($result as $font) {
		$fonts_list[] = $font;
	}
	echo implode('%7C', $fonts_list);
}
```

Should I stop worrying about that? Actually, if one uses a lot of ```google-fonts``` options (H1, H2, H3, H4, H5, H6, logo, title bar, footer headings, loop headings, body typography, etc...) it will print a long meta-link. But the content is the same.

Regards,

**Comment by valendesigns** [comment]
There are too many issues for a single ticket. Please open multiples. I've changed the title of this issue and included a solution in the 2.5.2 hotfix coming soon.
